homeassistant/components/plugwise/climate.py

```diff
@@ -238,62 +207,93 @@ def temperature_unit(self):
         """Return the unit of measured temperature."""
         return TEMP_CELSIUS
 
-    def set_temperature(self, **kwargs):
+    async def async_set_temperature(self, **kwargs):
         """Set new target temperature."""
-        _LOGGER.debug("Adjusting temperature")
         temperature = kwargs.get(ATTR_TEMPERATURE)
-        if temperature is not None and self._min_temp < temperature < self._max_temp:
-            _LOGGER.debug("Changing temporary temperature")
-            self._api.set_temperature(self._domain_objects, temperature)
+        if (temperature is not None) and (
+            self._min_temp < temperature < self._max_temp
+        ):
+            try:
+                await self._api.set_temperature(self._loc_id, temperature)
+                self._setpoint = temperature
+                self.async_write_ha_state()
+            except Smile.PlugwiseError:
+                _LOGGER.error("Error while communicating to device")
         else:
             _LOGGER.error("Invalid temperature requested")
 
-    def set_hvac_mode(self, hvac_mode):
+    async def async_set_hvac_mode(self, hvac_mode):
         """Set the hvac mode."""
-        _LOGGER.debug("Adjusting hvac_mode (i.e. schedule/schema)")
-        schema_mode = "false"
+        state = "false"
```

Comment:
Its a bit odd that it takes the magic string `"false"` here.  Can we at least make these constants ?

Reply:
Added SCHEDULED_{ON|OFF} as CONSTs